Build fails in the extension

### Describe the bug

GH actions due the build of the tests in the `extension/action-token-authenticator` fails with the error like:

```
[INFO] 3 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/mposolda/IdeaProjects/keycloak-quickstarts/extension/action-token-authenticator/src/test/java/org/keycloak/quickstart/ArquillianActionTokenWithAuthenticatorTest.java:[128,116] incompatible types: com.google.common.collect.ImmutableMap<java.lang.String,java.lang.String> cannot be converted to java.util.Map<java.lang.String,java.lang.Object>
[ERROR] /home/mposolda/IdeaProjects/keycloak-quickstarts/extension/action-token-authenticator/src/test/java/org/keycloak/quickstart/ArquillianActionTokenWithAuthenticatorTest.java:[139,19] incompatible types: com.google.common.collect.ImmutableMap<java.lang.String,java.lang.String> cannot be converted to java.util.Map<java.lang.String,java.lang.Object>
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.684 s

```

### Version

nightly from 2024-06-27

### Expected behavior

All quickstarts are OK and GH actions are working

### Actual behavior

GH actions are not OK

### How to Reproduce?

_No response_

### Anything else?

_No response_

--- a/.github/scripts/start-local-server.sh
+++ b/.github/scripts/start-local-server.sh
@@ -20,6 +20,9 @@ if [ "$1" = "extension" ]; then
   # extensions/event-store-mem example
   SERVER_ARGS="$SERVER_ARGS --spi-events-store-provider=in-mem";
 
+  # extension/extend-admin-console-spi
+  SERVER_ARGS="$SERVER_ARGS --features=declarative-ui";
+
   if [ "$1" == "extension" ]; then
     mvn -Dextension -DskipTests -B -Dnightly clean package
     cp extension/action-token-authenticator/target/action-token-example.jar $dist/providers

--- a/extension/action-token-authenticator/src/test/java/org/keycloak/quickstart/ArquillianActionTokenWithAuthenticatorTest.java
+++ b/extension/action-token-authenticator/src/test/java/org/keycloak/quickstart/ArquillianActionTokenWithAuthenticatorTest.java
@@ -18,7 +18,6 @@
 
 package org.keycloak.quickstart;
 
-import com.google.common.collect.ImmutableMap;
 import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.arquillian.container.test.api.OperateOnDeployment;
 import org.jboss.arquillian.container.test.api.TargetsContainer;
@@ -56,6 +55,7 @@ import java.io.IOException;
 import java.net.URL;
 import java.time.Duration;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.concurrent.TimeUnit;
 
@@ -125,7 +125,7 @@ public class ArquillianActionTokenWithAuthenticatorTest {
         importTestRealm("admin", "admin", "/quickstart-realm.json");
 
         // Update authentication flow to use external application redirection
-        qsRealm.flows().copy("browser", ImmutableMap.<String, String>builder().put("newName", "browser-copy").build()).close();
+        qsRealm.flows().copy("browser", Map.of("newName", "browser-copy")).close();
         List<AuthenticationExecutionInfoRepresentation> executions = qsRealm.flows().getExecutions("browser-copy");
 
         AuthenticationExecutionInfoRepresentation browserFormFlow = executions.stream()
@@ -134,9 +134,8 @@ public class ArquillianActionTokenWithAuthenticatorTest {
           .orElseThrow(() -> new AssertionError("Could not find forms flow"));
 
         qsRealm.flows().addExecution(browserFormFlow.getDisplayName(),
-          ImmutableMap.<String, String>builder()
-            .put("provider", ExternalAppAuthenticatorFactory.ID)
-            .build());
+                Map.of("provider", ExternalAppAuthenticatorFactory.ID)
+        );
 
         executions = qsRealm.flows().getExecutions("browser-copy");
         AuthenticationExecutionInfoRepresentation extAppExecution = executions.stream()

--- a/extension/extend-admin-console-spi/README.md
+++ b/extension/extend-admin-console-spi/README.md
@@ -21,7 +21,13 @@ Run this example by compiling with maven:
 mvn -Pextension clean install -DskipTests=true
 ```
 
-Then copy the generated `extend-admin-ui.jar` into your <keycloak-server>/providers folder and run the server using `kc.[sh|bat] start-dev`.
+Then copy the generated `extend-admin-ui.jar` into your <keycloak-server>/providers folder and run the server using:
+
+```bash
+kc.[sh|bat] start-dev --features=declarative-ui
+```
+
+NOTE: In order to properly use the SPI, the `declarative-ui` feature must be turned on.
 
 ### Screenshots
 
@@ -38,7 +44,7 @@ After you created an item you can view the details by clicking on it in the list
 Make sure you have a Keycloak server running with an admin user in the master realm and copy the generated jar file into the providers folder.
 Your Keycloak server should be listening on `http://localhost:8180`. You can achieve this by running:
 ```
-./kc.sh start-dev --http-port=8180
+./kc.sh start-dev --http-port=8180 --features=declarative-ui
 ```
 
 You need to have Chrome browser installed and updated to the latest version. See the root README for additional details.

--- a/extension/extend-admin-console-spi/src/test/java/org/keycloak/quickstart/ExtendAdminConsoleTest.java
+++ b/extension/extend-admin-console-spi/src/test/java/org/keycloak/quickstart/ExtendAdminConsoleTest.java
@@ -88,7 +88,7 @@ public class ExtendAdminConsoleTest {
         adminConsole.navigateTo();
         waitForPageToLoad();
         loginPage.login("admin", "admin");
-        assertThat(webDriver.getTitle(), containsString("Keycloak Administration UI"));
+        assertThat(webDriver.getTitle(), containsString("Keycloak Administration Console"));
 
         Assert.assertTrue(adminConsole.isTodoMenuPresent());
         adminConsole.clickTodoMenuItem();

--- a/extension/user-storage-simple/README.md
+++ b/extension/user-storage-simple/README.md
@@ -67,6 +67,13 @@ admin console or on the registration pages of <span>Keycloak</span>, will be cre
 to the Users tab in the Admin Console and create a new user, you'll be able to see the provider in action.  You can also
 edit the file yourself to add the username/password pairs you want.
 
+Disable user profile attributes
+-------------------------------
+Because the user-storage providers in this application use just attribute `username`, it may be needed to disable attributes `email`, `firstName` and `lastname` in the user-profile configuration
+to not be required or even visible. After login to admin console, you can go to tab `Realm settings` -> tab `User Profile` -> Click to requested attribute (like EG. `email`) and uncheck `Permissions`
+for `User` for both `edit` and `view`. This step is not needed in `master` realm, but may be needed in any other realm as attributes `email`, `firstName`, `lastname` and visible and required by default. 
+
+
 Integration test of the Quickstart
 ----------------------------------
 

--- a/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/ArquillianSimpleStorageTest.java
+++ b/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/ArquillianSimpleStorageTest.java
@@ -28,12 +28,17 @@ import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.keycloak.common.util.MultivaluedHashMap;
+import org.keycloak.models.UserModel;
 import org.keycloak.quickstart.page.ConsolePage;
 import org.keycloak.representations.idm.ComponentRepresentation;
 import org.keycloak.representations.idm.RequiredActionProviderRepresentation;
 import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.representations.userprofile.config.UPAttributePermissions;
+import org.keycloak.representations.userprofile.config.UPAttributeRequired;
+import org.keycloak.representations.userprofile.config.UPConfig;
 import org.keycloak.test.FluentTestsHelper;
 import org.keycloak.test.page.LoginPage;
+import org.keycloak.userprofile.config.UPConfigUtils;
 import org.openqa.selenium.WebDriver;
 
 import java.io.IOException;
@@ -92,8 +97,26 @@ public class ArquillianSimpleStorageTest {
         RequiredActionProviderRepresentation ra = r.getTestRealmResource().flows().getRequiredAction("VERIFY_PROFILE");
         ra.setEnabled(false);
         r.getTestRealmResource().flows().updateRequiredAction("VERIFY_PROFILE", ra);
+        disableUserProfileAttributes(r);
         webDriver.manage().timeouts().pageLoadTimeout(60, TimeUnit.SECONDS);
         webDriver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+    }
+
+    // Disable email, firstName, lastName attributes from user-profile
+    private void disableUserProfileAttributes(FluentTestsHelper r) {
+        UPConfig upConfig = r.getTestRealmResource().users().userProfile().getConfiguration();
+
+        removeUserPermissionsFromAttribute(upConfig, UserModel.EMAIL);
+        removeUserPermissionsFromAttribute(upConfig, UserModel.FIRST_NAME);
+        removeUserPermissionsFromAttribute(upConfig, UserModel.LAST_NAME);
+
+        r.getTestRealmResource().users().userProfile().update(upConfig);
+    }
+
+    private void removeUserPermissionsFromAttribute(UPConfig upConfig, String attrName) {
+        UPAttributePermissions upAttributePermissions = upConfig.getAttribute(attrName).getPermissions();
+        upAttributePermissions.getEdit().remove(UPConfigUtils.ROLE_USER);
+        upAttributePermissions.getView().remove(UPConfigUtils.ROLE_USER);
     }
 
     @After

--- a/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/page/ConsolePage.java
+++ b/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/page/ConsolePage.java
@@ -10,10 +10,10 @@ import static org.jboss.arquillian.graphene.Graphene.waitAjax;
  */
 public class ConsolePage {
 
-    @FindBy(xpath = "//a[text() = 'Sign out']")
+    @FindBy(xpath = "//span[text() = 'Sign out']")
     private WebElement logoutLink;
 
-    @FindBy(xpath = "//div[@data-testid='options']")
+    @FindBy(xpath = "//button[@data-testid='options-toggle']")
     private WebElement profileMenu;
 
     public void logout() {

--- a/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/util/StorageManager.java
+++ b/extension/user-storage-simple/src/test/java/org/keycloak/quickstart/util/StorageManager.java
@@ -40,6 +40,7 @@ public class StorageManager {
     }
 
     public static void deleteStorage() {
+        if (file == null) return;
         try {
             Files.deleteIfExists(file);
         } catch (IOException e) {

